libraries/Ethernet/src/ETH.cpp

```diff
@@ -1037,6 +1060,18 @@ uint8_t ETHClass::linkSpeed() const {
   return (link_speed == ETH_SPEED_10M) ? 10 : 100;
 }
 
+bool ETHClass::setLinkSpeed(uint16_t speed) {
+  if (_eth_handle == NULL) {
+    return false;
+  }
+  eth_speed_t link_speed = (speed == 10)?ETH_SPEED_10M:ETH_SPEED_100M;
+  esp_err_t err = esp_eth_ioctl(_eth_handle, ETH_CMD_S_SPEED, &link_speed);
+  if (err != ESP_OK) {
+    log_e("Failed to set link speed: 0x%x: %s", err, esp_err_to_name(err));
+  }
```

Comment:
Adding a log_v or log_i might be good, if someone select other speed then 10, to see what link speed is used (100).

Reply:
I left it as-is because I do not expect this to ever be used and also because I expect 1Gbps support to come soon

Reply:
ok, so later on maybe a check if its 10,100 or 1000 might be good.